app/src/main/java/com/bogdan/codeforceswatcher/features/auth/RestorePasswordComposeActivity.kt

```diff
@@ -139,20 +149,13 @@ class RestorePasswordComposeActivity : ComponentActivity(), StoreSubscriber<Auth
 
     override fun onStop() {
         super.onStop()
-        store.dispatch(AuthRequests.ResetRestorePasswordMessage)
+        resetRestorePasswordMessage()
         store.unsubscribe(this)
     }
 
-    private fun forgotPassword(email: String) {
-        store.dispatch(AuthRequests.SendPasswordReset(email))
-    }
-
-    private fun startRestorePasswordMailSentActivity() {
-        startActivity(Intent(this, RestorePasswordMailSentComposeActivity::class.java))
-    }
-
     override fun onNewState(state: AuthState) {
-        if (state.status == AuthState.Status.DONE) startRestorePasswordMailSentActivity()
         authState.postValue(state)
+        if (state.status == AuthState.Status.DONE) startRestorePasswordMailSentActivity()
```

Comment:
When you have multiple `if`s on the same property, it's recommended to convert to `when`.